src/amber/cli/templates/auth/src/models/{{name}}.cr.ecr

```diff
@@ -22,16 +12,34 @@ class <%= class_name %> < Granite::ORM::Base
   timestamps
 <% end -%>
 
-  def encrypt_password
-    @encrypted_password = Crypto::Bcrypt::Password.create(@password.not_nil!, cost: 10).to_s
+  validate :email, "is required", -> (<%= @name %> : <%= class_name %>) do
+    (email = <%= @name %>.email) ? !email.empty? : false
+  end
+
+  validate :password, "is too short", -> (<%= @name %> : <%= class_name %>) do
+    <%= @name %>.password_changed? ? <%= @name %>.valid_password_size? : true
+  end
+
+  def password=(password)
+    @new_password = password
+    @hashed_password = Bcrypt::Password.create(password, cost: 10).to_s
+  end
+
+  def password
+    @password ||= (hash = hashed_password) ? Bcrypt::Password.new(hash) : nil
```

Comment:
Will the `||=` cause any issues if they change their password?  Should this reset when they set a new password?

Reply:
That's a good point.

Normally with the flow through the program, `password=` is called before `password` but there isn't anything stopping it from happening the other way around.

I don't think you have the blowfish overhead with `Bcrypt::Password.new(hash)` only on `Bcrypt::Password.create` and `Bcrypt::Password#==`

I was thinking of ways to still memoize `@password` but I'm not sure it is even worth it.